### PR TITLE
Use webmock for tests

### PR DESCRIPTION
**Pull request by MatMoore**
Hi @dkhamsing 

Because of the test issues I found in https://github.com/dkhamsing/awesome_bot/pull/172 I had ago at rewriting some of them. 

This uses webmock to stub out HTTP requests instead of actually requesting real websites. What do you think of this approach?

I only changed a few broken ones for now, but I noticed other tests that are quite flaky at the moment because the servers they hit don't behave consistently.

For example, if I query the httpbin site (used in check_spec.rb) through curl, every now and again it will serve a 301 response instead of a 200:

```
$ curl -i 'http://user:passwd@httpbin.org/basic-auth/user/passwd'
HTTP/1.1 200 OK
Connection: keep-alive
Server: gunicorn/19.8.1
Date: Thu, 05 Jul 2018 15:11:28 GMT
Content-Type: application/json
Content-Length: 37
Access-Control-Allow-Origin: *
Access-Control-Allow-Credentials: true
Via: 1.1 vegur

{"authenticated":true,"user":"user"}
$ curl -i 'http://user:passwd@httpbin.org/basic-auth/user/passwd'
HTTP/1.1 301 Moved Permanently
Date: Thu, 05 Jul 2018 15:11:29 GMT
Content-Type: text/plain
Transfer-Encoding: chunked
Connection: keep-alive
Location: https://httpbin.org/basic-auth/user/passwd
x-now-trace: bru1
server: now
cache-control: max-age=0
```

### CODE REVIEW

**Comment by dkhamsing** [comment]
That’s a big change, can you summarize the prod and cons ?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by MatMoore** [comment]
The pro is that the tests always have the same setup, so you know if they fail, then it's because something is wrong with the actual code being tested. Whereas right now I can run the test twice in succession and get different results (see the example above for why).

There aren't much cons except as you say it's a big change. It's also confusing if some of the tests talk to external websites and some don't, so I would prefer to update all the specs rather than just the ones I've added.

**Comment by dkhamsing** [comment]
Will review more and think on this, thank you!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by MatMoore** [comment]
Since I took a different approach in https://github.com/dkhamsing/awesome_bot/pull/172, I'll close this PR for now.

**Comment by dkhamsing** [comment]
🙇‍♂️
